segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerTests.java

```diff
@@ -1539,6 +1542,31 @@ private void activateAllSegments(Collection<String> segmentNames, TestContext co
         return container.read(segmentName, 0, 1, TIMEOUT).thenAccept(ReadResult::close);
     }
 
+    /**
+     * Blocks until all operations processed so far have been added to the ReadIndex and InMemoryOperationLog.
+     * This is needed to simplify test verification due to the fact that the the OperationProcessor commits operations to
+     * the ReadIndex and InMemoryOperationLog asynchronously, after those operations were ack-ed. This method makes use
+     * of the fact that the OperationProcessor/MemoryStateUpdater will still commit such operations in sequence; it
+     * creates a new segment, writes 1 byte to it and issues a read (actual/future) and waits until it's completed - when
+     * it is, it is guaranteed that everything prior to that has been committed.
+     */
+    @SneakyThrows
```

Comment:
Can we narrow this? It looks like the get() at the end should be: using handleInterrupted. So it might be neater to package this as Futures.getAndHandleExceptions()

Reply:
There are 3 Exceptions being thrown in this class. Since this is a unit test, I just have the method declare `throws Exception` and let it bubble up.